rest/route/ip_address.go

```diff
@@ -0,0 +1,58 @@
+package route
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+)
+
+////////////////////////////////////////////////////////////////////////
+//
+// GET /rest/v2/hosts/ip_address/{ip_address}
+
+type addressGetHandler struct {
+	IP   string
+	Host *host.Host
+	sc   data.Connector
+}
+
+func makeGetHostByIP(sc data.Connector) gimlet.RouteHandler {
+	return &addressGetHandler{
+		sc: sc,
+	}
+}
+
+func (h *addressGetHandler) Factory() gimlet.RouteHandler {
+	return &addressGetHandler{
+		sc: h.sc,
+	}
+}
+func (h *addressGetHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.IP = gimlet.GetVars(r)["ip_address"]
+
+	if h.IP == "" {
+		return errors.New("request data incomplete")
```

Comment:
I'm afraid that a user might not realize immediately what it is that is incomplete about the request data. Can we perhaps add a bit more explanation?

Reply:
done

Reply:
this is better. I think it might be even better phrased as "ip_address must be specified" or "ip_address cannot be empty" so that it is clearer to the user that they can fix the error by fixing the input. 

It might also be a good idea to wrap it in gimlet.ErrorResponse with a http.StatusBadRequest status code.

Reply:
done